spec/spec_helper.rb

```diff
@@ -13,6 +13,7 @@
                                                                              # Original fact sources:
 add_custom_fact :puppet_environmentpath, '/etc/puppetlabs/code/environments' # puppetlabs-stdlib
 add_custom_fact :root_home, '/root'                                          # puppetlabs-stdlib
+add_custom_fact :service_provider, 'redhat'                                  # puppetlabs-stdlib
```

Comment:
I'd say `systemd` is a more common default if I look at `metadata.json` but if you really want a correct implementation (from https://github.com/voxpupuli/modulesync_config/pull/431):

```ruby
# Rough conversion of grepping in the puppet source:
# grep defaultfor lib/puppet/provider/service/*.rb
add_custom_fact :service_provider, ->(os, facts) do
  case facts[:osfamily].downcase
  when 'archlinux'
    'systemd'
  when 'darwin'
    'launchd'
  when 'debian'
    if facts[:operatingsystem] == 'Ubuntu'
      facts[:operatingsystemmajrelease].to_i >= 15 ? 'systemd' : 'upstart'
    elsif facts[:operatingsystem] == 'Debian' && facts[:operatingsystemmajrelease].to_i >= 8
      'systemd'
    else
      'debian'
    end
  when 'freebsd'
    'freebsd'
  when 'gentoo'
    'openrc'
  when 'openbsd'
    'openbsd'
  when 'redhat'
    facts[:operatingsystemrelease].to_i >= 7 ? 'systemd' : 'redhat'
  when 'suse'
    facts[:operatingsystemmajrelease].to_i >= 12 ? 'systemd' : 'redhat'
  when 'windows'
    'windows'
  else
    'init'
  end
end
```

Reply:
Oh wow, this is amazing! Switched to this way more complete implementation.